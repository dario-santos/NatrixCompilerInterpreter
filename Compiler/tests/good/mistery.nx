function mistery(n : int) : int
{
    type a : array 10 of [0..maxint];
    val v : a filled by 0;
    
    foreach i in [0 .. 9]
    {
        v[i] := i;
    }

    if(n < 10)
    {
        return v[n];
    }
        
    return (n / 10) + v[n % 10];
}

printn(mistery(0));
printn(mistery(42));
printn(mistery(1024));
