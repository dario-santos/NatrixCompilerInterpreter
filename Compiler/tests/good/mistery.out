0
6
106
